Source/PeleLMeX_Plot.cpp

```diff
@@ -1052,30 +1045,33 @@ PeleLM::initLevelDataFromPlt(int a_lev, const std::string& a_dataPltFile)
         SootConst sc;
         amrex::Real* momV = sc.MomOrderV.data();
         amrex::Real* momS = sc.MomOrderS.data();
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
-#endif
-        for (amrex::MFIter mfi(ldata_p->state, amrex::TilingIfNotGPU());
-             mfi.isValid(); ++mfi) {
-          const amrex::Box& bx = mfi.tilebox();
-          auto const& soot_arr = ldata_p->state.array(mfi, FIRSTSOOT);
-          amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-              for (int n = 0; n < NUM_SOOT_MOMENTS; n++) {
-                amrex::Real soot_exp = 3. - (3. * momV[n] + 2. * momS[n]);
-                soot_arr(i, j, k, n) *= std::pow(100., soot_exp);
-              }
-              soot_arr(i, j, k, NUMSOOTVAR - 1) *= 1.E6;
-            });
+        auto const& state_ma = ldata_p->state.arrays();
+        amrex::Real soot_exp[NUM_SOOT_MOMENTS] = {0.0};
+        for (int n = 0; n < NUM_SOOT_MOMENTS; ++n) {
+          soot_exp[n] = 3. - (3. * momV[n] + 2. * momS[n]);
         }
+        amrex::ParallelFor(
+          ldata_p->state, amrex::IntVect(0), NUM_SOOT_MOMENTS,
+          [state_ma, soot_exp] AMREX_GPU_DEVICE(
+            int box_no, int i, int j, int k, int n) noexcept {
+            amrex::Array4<amrex::Real> soot(state_ma[box_no], FIRSTSOOT);
+            soot(i, j, k, n) *= std::pow(100., soot_exp[n]);
+          });
+        amrex::Gpu::streamSynchronize();
+        ldata_p->state.mult(1.E6, FIRSTSOOT + NUM_SOOT_MOMENTS - 1, 1);
```

Comment:
I think this line should have `FIRSTSOOT + NUMSOOTVAR - 1` rather than `FIRSTSOOT + NUM_SOOT_MOMENTS - 1` in order to match the previous implementation

Reply:
I've fixed, but also just combined it into the ParallelFor above for similar reasons as the HRR one from earlier.